modules/rmarkdown/main.nf

```diff
@@ -0,0 +1,91 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+include { dump_params_yml } from "./parametrize"
+
+params.options = [:]
+options        = initOptions(params.options)
+params.parametrize = true
+params.implicit_params = true
+params.meta_params = true
+
+process RMARKDOWN {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    //NB: You likely want to override this with a container containing all required
+    //dependencies for you analysis. The container at least needs to contain the
+    //yaml and rmarkdown R packages.
+    conda (params.enable_conda ? "r-tidyverse=1.2.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/r-tidyverse:1.2.1"
+    } else {
+        container "quay.io/biocontainers/r-tidyverse:1.2.1"
+    }
+
+    input:
+    tuple val(meta), path(notebook)
+    val(parameters)
+    path(input_files)
+
+    output:
+    tuple val(meta), path("*.html"), emit: report
+    path("artifacts/*"), emit: artifacts, optional: true
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+
```

Comment:
check the rmd file contain sessionInfo section or not. if not add seesionInfo() at the end of rmd file.

Reply:
While I like the idea of storing `sessionInfo`, I'm not a big fan of modifying the notebook. But maybe we can separately dump the `sessionInfo` into a log file.

Reply:
sessionInfo might be overkill just for the rendering the notebook itself.

https://github.com/nf-core/modules#general

```
If the software is unable to output a version number on the command-line then a variable called VERSION can be manually specified to create this file e.g. homer/annotatepeaks module.
```
Might be sufficient? Or a some custom Rscript command to load the  two packages and then run `sessionInfo()` and 
then some bash cleanup?

Half completed example:

```
R -e "library('rmarkdown')" -e "library('yaml')" -e 'sessionInfo()' | grep -e 'R version' -e 'other' -A 1
```

Reply:
Ok, I added a sessioninfo log as separate output. 
Into the version.yml/versions.txt, I will only include the `rmarkdown` and `yaml` packages which are directly needed for rendering the notebook.